qiime2/sdk/action.py

```diff
@@ -409,7 +419,8 @@ def _run_parsl_action(action, ctx, execution_ctx, mapped_args,
             # in the right cache
             with ctx.cache:
                 exe = action._bind(
-                    lambda: qiime2.sdk.Context(parent=ctx), execution_ctx)
+                    lambda: ctx if ctx._parent is None
+                    else qiime2.sdk.Context(parent=ctx), execution_ctx)
```

Comment:
It isn't clear when the `_parent` isn't `None`.  From poking around it seems like perhaps everything ends up in the same parent context? I guess that could work, now that destructors don't exist.

Reply:
It is possible we have ended up in that state yeah... And yeah I'm not sure if we care about that if Context is no longer cleaning up

Reply:
Ok so we are setting parent context's when we call an action inside of a pipeline. This is used to identify when we have hit the end of the root pipeline so we know when to wait for our results.